docs/introduction.rst

```diff
@@ -0,0 +1,111 @@
+the AI. Figure 3 shows sample saliency maps for text and images.
+
+.. figure:: figures/intro-fig-03.png
+
+*Figure 3: Sample saliency maps for (left) text (from* `Tuckey et al.
+<https://arxiv.org/abs/1907.05664>`_) *and (right) images (from* `Dong et
+al. <https://openaccess.thecvf.com/content_CVPRW_2019/html/Explainable_AI/Dong_Explainability_for_Content-Based_Image_Retrieval_CVPRW_2019_paper.html>`_
+*)*
+
+.. note:: The XAITK-Saliency toolkit focuses on providing saliency
+          maps for images.
+
+Image Saliency Maps: An Intuitive Introduction
+----------------------------------------
+
+Figure 4 shows a deep learning pipeline for recognizing objects in
+images; pixels in the image (in green) are processed by the pipeline
+to produce the output (in orange). Here, the system has been trained
+to recognize 1000 object types. Its output is a list of 1000 numbers,
+one for each object type; each number is between 0 and 1, representing
+the system's estimate of whether or not that particular object is in
+the image.
+
+.. figure:: figures/cnn-tagged-scaled.png
+
+*Figure 4: Typical object recognition CNN architecture; the image
+(green, left) is processed left to right through the CNN to produce an
+output vector (orange, right). Diagram from* `Krizhevsky et
+al. <https://proceedings.neurips.cc/paper/4824-imagenet-classification-with-deep-convolutional-neural-networks.pdf>`_
+
+
+This operation is "all or nothing" at both ends: the entire image must
+be processed, and the the entire vector must be output. There is no
+mechanism by which a subset of the output can be traced back to a
+subset of the input. Yet it seems reasonable to ask questions such as:
+
+* The input image is a camel; why did the system give a higher
+  likelihood to "horse" than "camel"?
+
+* The input image is a kitchen, but the system gave a high likelihood
+  to "beach ball". What parts of the image were responsible for this?
+
+* The input image contains two dogs, and the system gave a high
+  likelihood for "dog". How would this change if one of the dogs
+  wasn't in the image?
+
+* The input image contains a dog and a cat, and the system gave a high
+  likelihood for "cat" but not "dog". How will the system respond if
+  the cat is removed?
+
+At some level, these questions require a degree of *introspection*;
+the system must produce not only the output, but also some information
+about **how** the output was produced. There are two popular
+approaches to this:
+
+1) The **white box** approach: the system is altered to open or expose
+   the internal state of the model. This state is examined while the
+   system is generating the output.
+
+2) The **black box** approach: the model is not opened; instead, we
+   create a series of *related images* which perturb or change the
+   original image in some way. By comparing the original output to the
+   output for the related images, we can deduce certain aspects of the
+   model's behavior.
```

Comment:
Do want our verbiage to be specific to images? If not we might want to change the use of "image" here to a more generic "input."

Reply:
sounds good